src/nvim/mark.c

```diff
@@ -378,8 +378,8 @@ pos_T *getmark_buf_fnum(buf_T *buf, int c, int changefile, int *fnum)
     }
   } else if (ASCII_ISLOWER(c)) {      /* normal named mark */
     posp = &(buf->b_namedm[c - 'a']);
-  } else if (ASCII_ISUPPER(c) || VIM_ISDIGIT(c)) {    /* named file mark */
-    if (VIM_ISDIGIT(c))
+  } else if (ASCII_ISUPPER(c) || isdigit(c)) {    /* named file mark */
```

Comment:
> It looks like there would be a lot of duplication if they were separated out like this:

I checked the surrounding code again, makes sense, ack.

> so It'd probably be worthwhile to deal with that given the platforms we support are relatively sane.

As part of another PR, I guess?